From 83d1a8cb3344ac16dd1b822528d73dc20c54397b Mon Sep 17 00:00:00 2001
From: Brandon Li <brandonli@apache.org>
Date: Wed, 1 Apr 2015 17:20:24 -0700
Subject: [PATCH 0810/1023] HDFS-8001 RpcProgramNfs3 : wrong parsing of
 dfs.blocksize. Contributed by Remi Catherinot

(cherry picked from commit 4d14816c269f110445e1ad3e03ac53b0c1cdb58b)
(cherry picked from commit 27970b3f46674ef7824a4478141d968ef2c16ebb)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: I90a462b44e3b24089501f44ab5dea7770fcf38c9
(cherry picked from commit cd5b6b6beb4ce6ce3d72819f80e2af0377262a62)
---
 .../hadoop/hdfs/nfs/nfs3/RpcProgramNfs3.java       |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs-nfs/src/main/java/org/apache/hadoop/hdfs/nfs/nfs3/RpcProgramNfs3.java b/hadoop-hdfs-project/hadoop-hdfs-nfs/src/main/java/org/apache/hadoop/hdfs/nfs/nfs3/RpcProgramNfs3.java
index aaf8135..a109315 100644
--- a/hadoop-hdfs-project/hadoop-hdfs-nfs/src/main/java/org/apache/hadoop/hdfs/nfs/nfs3/RpcProgramNfs3.java
+++ b/hadoop-hdfs-project/hadoop-hdfs-nfs/src/main/java/org/apache/hadoop/hdfs/nfs/nfs3/RpcProgramNfs3.java
@@ -186,7 +186,7 @@ public RpcProgramNfs3(NfsConfiguration config, DatagramSocket registrationSocket
     clientCache = new DFSClientCache(config);
     replication = (short) config.getInt(DFSConfigKeys.DFS_REPLICATION_KEY,
         DFSConfigKeys.DFS_REPLICATION_DEFAULT);
-    blockSize = config.getLong(DFSConfigKeys.DFS_BLOCK_SIZE_KEY,
+    blockSize = config.getLongBytes(DFSConfigKeys.DFS_BLOCK_SIZE_KEY,
         DFSConfigKeys.DFS_BLOCK_SIZE_DEFAULT);
     bufferSize = config.getInt(
         CommonConfigurationKeysPublic.IO_FILE_BUFFER_SIZE_KEY,
-- 
1.7.9.5

